### PR TITLE
[Dawn] Peripheral Display Filter

### PR DESCRIPTION
**Pull request by kevinmasd**
TeamFlag has been edited out of peripheral display. It had been conflicting with any other sensor in the default case (i.e. not in `cleanerNames`).

Additionally, can somebody check if all the displayable peripherals are enumerated in `cleanerNames`?

### CODE REVIEW

**Comment by yizow** [comment]
I'm not quite sure what this is supposed to do.
Currently, team flag does not show up as a sensor.

This pull request... also makes it not show up?

**Comment by kevinmasd** [comment]
What's happening currently is that the accordion/panel for ExampleSensor and TeamFlag are conflicting one another, since Dawn doesn't recognize either. As a result, TeamFlag would never be seen because Electron only displays the panel that came earliest (alphabetically being ExampleSensor).

File: dawn/renderer/components/PeripheralList.js [hunk 1]
```diff
@@ -18,13 +18,18 @@ cleanerNames[PeripheralTypes.ServoControl] = 'Servo Controllers';
 cleanerNames[PeripheralTypes.RFID] = 'RFID';
 cleanerNames[PeripheralTypes.YogiBear] = 'Yogi Bear';
 
+const filter = [PeripheralTypes.TeamFlag];
```
Comment (kevinji) [inline dawn/renderer/components/PeripheralList.js]: Can you make this a set (`new Set([PeripheralTypes.TeamFlag])`)?

File: dawn/renderer/components/PeripheralList.js [hunk 2]
```diff
@@ -18,13 +18,18 @@ cleanerNames[PeripheralTypes.ServoControl] = 'Servo Controllers';
...
+
 const handleAccordion = (array) => {
   const peripheralGroups = {};
+
   array.forEach((p) => {
```
Comment (kevinji) [inline dawn/renderer/components/PeripheralList.js]: You can extract the filter includes check with
`array.filter(p => !filter.has(p.device_type)).forEach(p => /* ... */)`
assuming you converted the filter to a Set.

**Review by kevinji — CHANGES_REQUESTED** [review]
Really minor stylistic differences.

**Comment by kevinmasd** [comment]
Changes done.

**Review by kevinji — APPROVED** [review]

**Pull request merged**